### PR TITLE
expand multi-series charm Ids in charm-related

### PR DESCRIPTION
**Pull request by mhilton**
Make the /v4 version of charm-related response contain charm ids that always contain a series.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1400/
Test PASSed.

File: internal/v4/relations.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+	resp, err := h.getMetadataForEntities(entities, includes, req, usesInterface)
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+	return resp, nil
+}
+
+func (h *ReqHandler) getMetadataForEntities(entities []*mongodoc.Entity, includes []string, req *http.Request, includeEntity func(*mongodoc.Entity) bool) ([]params.MetaAnyResponse, error) {
+	// TODO(rog) make this concurrent.
```
Comment (rogpeppe) [inline internal/v4/relations.go]: This comment belongs in expandMultiSeries now, I think.

File: internal/v4/relations_test.go [hunk 2]
```diff
@@ -125,6 +145,12 @@ var metaCharmRelatedTests = []struct {
 		},
 		Requires: map[string][]params.MetaAnyResponse{
 			"http": {{
+				Id: charm.MustParseURL("precise/multi-series-1"),
```
Comment (rogpeppe) [inline internal/v4/relations_test.go:149]: // V4 SPECIFIC

**Comment by rogpeppe** [comment]
LGTM with a couple of trivial suggestions, thanks.

File: internal/v4/api.go [hunk 3]
```diff
@@ -247,20 +250,25 @@ func (h *ReqHandler) serveExpandId(id *router.ResolvedURL, w http.ResponseWriter
 }
 
 // expandMultiSeries calls the provided append function once for every
-// supported series of each entry in the given entities slice. The
-// series argument will be passed as that series and the doc argument
-// will point to the entity.
+// supported series of each entry in the given entities slice. The series
+// argument will be passed as that series and the doc argument will point
+// to the entity. This function will only return an error if the append
+// function returns an error, such an error will be returned without
```
Comment (frankban) [inline internal/v4/api.go]: s/,/; or : ?

**Comment by frankban** [comment]
:+1:

File: internal/v4/relations.go (hunk 1 above)
Comment (mhilton) [inline internal/v4/relations.go]: done

File: internal/v4/relations_test.go (hunk 2 above)
Comment (mhilton) [inline internal/v4/relations_test.go:149]: done

File: internal/v4/api.go (hunk 3 above)
Comment (mhilton) [inline internal/v4/api.go]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1401/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**